### PR TITLE
Add action and title to tree component

### PR DESCRIPTION
**Pull request by tpitts-CFI**
Task: Add action and title to Tree Component (e287e8f0-3836-11ec-976e-2df811d36180)
* Tree Component will indicate reference type and referencing item

### CODE REVIEW

**Comment by tpitts-CFI** [comment]
need to use customLabel instead of propertyName

**Review by gmckune — APPROVED** [review]

**Review by cfi-dphillips — CHANGES_REQUESTED** [review]

File: kohese-portal/client/src/components/compare-items/item-comparison/compare-items.component.ts [hunk 1]
```diff
@@ -178,6 +178,12 @@ export class CompareItemsComponent implements OnInit {
     this._dialogService.openComponentDialog(TreeComponent, {
       data: {
         root: TreeConfiguration.getWorkingTree().getRootProxy(),
+        getTitle: {
+          definition: {
+            name: 'compareItems',
+            reason: 'choosing an item to compare against'
```
[the remark this answers is not in the document]
Comment (tpitts-CFI) [reply kohese-portal/client/src/components/compare-items/item-comparison/compare-items.component.ts]: OBE

File: kohese-portal/client/src/components/object-editor/format-object-editor/field/field.class.ts [hunk 2]
```diff
@@ -245,6 +245,7 @@ export class Field {
     selection = await this._dialogService.openComponentDialog(TreeComponent, {
       data: {
         root: treeConfiguration.getRootProxy(),
+        getTitle: this._dataModel['classProperties'][attributeName],
```
[the remark this answers is not in the document]
Comment (tpitts-CFI) [reply kohese-portal/client/src/components/object-editor/format-object-editor/field/field.class.ts]: OBE

File: kohese-portal/client/src/components/tree/tree.component.ts [hunk 3]
```diff
@@ -157,6 +157,94 @@ export class TreeComponent implements OnInit, AfterViewInit, Dialog {
...
 
+  private _getTitle: string;
+  get getTitle() {
+    return this._getTitle;
+  }
+  @Input('getTitle')
+  set getTitle(getTitle: string) {
+
+    if(getTitle !== undefined) {
+      if(getTitle['definition']['name'] === 'parentId') {
```
[the remark this answers is not in the document]
Comment (tpitts-CFI) [reply kohese-portal/client/src/components/tree/tree.component.ts]: OBE

File: kohese-portal/client/src/components/import/import.component.html [hunk 4]
```diff
@@ -18,6 +18,7 @@
 <div [ngStyle]="{ 'height': isDialogInstance() ? '40%' : '45%' }" style="width: 100%;">
   <h3 style="height: 10%;">Parent</h3>
   <tree #itemProxyTree style="height: 87%;"
+    [isImport]="!!true"
```
Comment (cfi-dphillips) [inline kohese-portal/client/src/components/import/import.component.html:21]: What is the purpose of !!true, wouldn't that be true?

**Review by cfi-dphillips — APPROVED** [review]

File: kohese-portal/client/src/components/import/import.component.html (hunk 4 above)
Comment (cfi-dphillips) [reply kohese-portal/client/src/components/import/import.component.html:21]: It appears to be required to ensure that the return is a boolean true.

Comment (gmckune) [reply kohese-portal/client/src/components/import/import.component.html:21]: it wasn't being sent correctly when i just sent 'true'. so saying 'not, not true' made sure to type it as boolean and true.

**Pull request merged**